build_runner/lib/src/build_script_generate/bootstrap.dart

```diff
@@ -156,27 +156,40 @@ Future<int> _createSnapshotIfNeeded(Logger logger) async {
     }
   }
 
-  String stderr;
   if (!await snapshotFile.exists()) {
     var mode = stdin.hasTerminal
         ? ProcessStartMode.normal
         : ProcessStartMode.detachedWithStdio;
+    var hadStdOut = false;
     await logTimedAsync(logger, 'Creating build script snapshot...', () async {
       var snapshot = await Process.start(Platform.executable,
           ['--snapshot=$scriptSnapshotLocation', scriptLocation],
           mode: mode);
-      stderr = (await snapshot.stderr
-              .transform(utf8.decoder)
-              .transform(LineSplitter())
-              .toList())
-          .join('');
+      await Future.wait([
+        snapshot.stderr
+            .transform(utf8.decoder)
+            .transform(LineSplitter())
+            .listen((l) {
+          logger.warning('stderr: $l');
+        }).asFuture(),
+        snapshot.stdout
+            .transform(utf8.decoder)
+            .transform(LineSplitter())
+            .listen((l) {
+          hadStdOut = true;
+          logger.fine('stdout: $l');
+        }).asFuture(),
+      ]);
     });
+    if (hadStdOut) {
+      logger.info('There was output on stdout while compiling the build script '
+          'snapshot, run with `--verbose` to see the it\n.');
```

Comment:
Will they need to do a clean first or something? If it was successful but with output I don't think the next run with rebuild the snapshot will it?

Reply:
Hmmm right, they would have to clean. Should we suggest a `clean` do you think? Just not emit a message?

Reply:
I guess maybe suggest a `clean` for now, and file an issue to add a `--force-snapshot` flag.

Reply:
I don't think we actually want to add a `--force-snapshot` flag - just a normal clean in this scenario should be fine IMO.